### PR TITLE
Map and Generator data generalised

### PR DESCRIPTION
**Pull request by wdigger**

### CODE REVIEW

**Comment by wdigger** [comment]
@jonls Pls, take a look.

File: src/map.cc [hunk 1]
```diff
@@ -435,42 +443,30 @@ Map::init_dimensions() {
...
-      if (generator.get_resource_type(pos_) != MineralsNone) {
-        tiles[pos_].resource = (generator.get_resource_type(pos_) & 7) << 5 |
-          (generator.get_resource_amount(pos_) & 0x1f);
-      } else {
-        tiles[pos_].resource = generator.get_resource_amount(pos_);
-      }
-    }
-  }
+  memcpy(landscape_tiles, generator.get_landscape(),
+         rows * cols * sizeof(LandscapeTile));
```
Comment (jonls) [inline src/map.cc]: We don't use the accessor functions on `MapGenerate` anymore, `get_height()`, `get_type_up()`, etc.? Should these just be removed then?

**Comment by wdigger** [comment]
Those methods are used in test.

**Comment by jonls** [comment]
Yeah, but why not just access through LandscapeTile in the test?

**Comment by wdigger** [comment]
Let remains, later I want to do depth refactoring for network.

**Comment by jonls** [comment]
ok 👍

**Pull request merged**